src/components/select/select.component.ts

```diff
@@ -216,15 +216,23 @@ export default class SlSelect extends ShoelaceElement implements ShoelaceFormCon
   }
 
   private addOpenListeners() {
-    document.addEventListener('focusin', this.handleDocumentFocusIn);
-    document.addEventListener('keydown', this.handleDocumentKeyDown);
-    document.addEventListener('mousedown', this.handleDocumentMouseDown);
+    //
+    // Listen on the root node instead of the document in case the elements are inside a shadow root
+    //
+    // https://github.com/shoelace-style/shoelace/issues/1763
+    //
+    // const root = this.getRootNode();
```

Comment:
did you mean to comment out `this.getRootNode()`?

Reply:
Nope! I commented it for testing. Fixed now — thanks!